certification/internal/policy/types.go

```diff
@@ -1,38 +1,27 @@
 package policy
 
-type Definition struct {
-	ValidatorFunc func(string) (bool, error)
-	Metadata      Metadata
-	HelpText      HelpText
-}
-
-func (pd *Definition) Validate(image string) (bool, error) {
-	return pd.ValidatorFunc(image)
-}
-
-func (pd *Definition) Meta() Metadata {
-	return pd.Metadata
-}
-
-func (pd *Definition) Help() HelpText {
-	return pd.HelpText
-}
-
-type PolicyInfo struct {
-	Metadata `json:"metadata" xml:"metadata"`
-	HelpText `json:"helptext"`
+// Policy as an interface containing all methods necessary
+// to use and identify a given policy.
+type Policy interface {
+	// Validate whether the asset enforces the policy.
+	Validate(image string) (result bool, err error)
+	// return the name of the policy
+	GetName() string
+	// return the policy's metadata
+	GetMetadata() Metadata
+	// return the policy's help text
+	GetHelp() HelpText
 }
 
 // Metadata contains useful information regarding the policy
-// being enforced
 type Metadata struct {
 	Description      string `json:"description" xml:"description"`
 	Level            string `json:"level" xml:"level"`
 	KnowledgeBaseURL string `json:"knowledge_base_url,omitempty" xml:"knowledgeBaseURL"`
 	PolicyURL        string `json:"policy_url,omitempty" xml:"policyURL"`
 }
 
-// HelpText is the help message associated with any given policy.
+// HelpText is the help message associated with any given policy
 type HelpText struct {
 	Message    string `json:"message" xml:"message"`
 	Suggestion string `json:"suggestion" xml:"suggestion"`
```

Comment:
Nitpick - I believe the punctuation here is 👍  from godoc's perspective. I've been inconsistent in other places where I've missed punctuation (my apologies). Probably better to leave this in place, and maybe add it in where it's missing. You don't have to add other missing periods, but we can leave this one here.